basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -1151,53 +1151,94 @@ namespace basic_autonomy
                 throw std::invalid_argument("Could not fit a spline curve along the given trajectory!");
             }
             RCLCPP_DEBUG_STREAM(rclcpp::get_logger(BASIC_AUTONOMY_LOGGER), "Got fit");
-            std::vector<lanelet::BasicPoint2d> all_sampling_points;
-            all_sampling_points.reserve(1 + future_geom_points.size() * 2);
 
+            // Add current vehicle position to front of future geometry points
             lanelet::BasicPoint2d current_vehicle_point(state.x_pos_global, state.y_pos_global);
-
-            future_geom_points.insert(future_geom_points.begin(),
-                                       current_vehicle_point); // Add current vehicle position to front of future geometry points
-
+            future_geom_points.insert(future_geom_points.begin(), current_vehicle_point);
             final_actual_speeds.insert(final_actual_speeds.begin(), state.longitudinal_vel);
 
-            //Compute points to local downtracks
-            std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(future_geom_points);
-
-            auto total_step_along_curve = static_cast<int>(downtracks.back() /detailed_config.curve_resample_step_size);
-
-            double scaled_steps_along_curve = 0.0; //from 0 (start) to 1 (end) for the whole trajectory
-
-            for(int steps_along_curve = 0; steps_along_curve < total_step_along_curve; steps_along_curve++){
-                lanelet::BasicPoint2d p = (*fit_curve)(scaled_steps_along_curve);
+            // Compute downtracks for original points (for interpolation reference)
+            std::vector<double> original_downtracks = carma_wm::geometry::compute_arc_lengths(future_geom_points);
+
+            // Now create resampled points using the spline
+            auto total_step_along_curve = static_cast<int>(original_downtracks.back() / detailed_config.curve_resample_step_size);
+            if (total_step_along_curve == 0) {
+                RCLCPP_WARN_STREAM(rclcpp::get_logger(BASIC_AUTONOMY_LOGGER),
+                "Available distance to resample is less than curve_resample_step_size. "
+                "Only considering the last point of the target destination to generate trajectory."
+                );
+                total_step_along_curve = 1; // This also avoids division by zero
+            }
 
-                all_sampling_points.push_back(p);
+            std::vector<lanelet::BasicPoint2d> resampled_points;
+            resampled_points.reserve(total_step_along_curve);
 
+            double scaled_steps_along_curve = 0.0; // from 0 (start) to 1 (end) for the whole trajectory
+            for(int step = 0; step < total_step_along_curve; step++){
                 scaled_steps_along_curve += 1.0 / total_step_along_curve;
```

Comment:
This should probably be incremented by `(step) / (total_step_along_curve - 1)`

**With current method** 
for step = 0 -> 10:
scaled_steps range from (0.1 -> 0.9)

**With suggested update**
for step = 0->10:
scaled_steps range from (0 ->1)

This is probably why the vehicle current_pose was being added back to the resampled point?

Reply:
It was actually missign the last point i think (until 0.9). So I included the last point